handlers/files.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const fs = require('fs');
+
+/**
+ * Reads all files and directories inside the next's pages folder and maps them into urls.
+ *
+ * @param {Array} sitemapEntries - An array of entries already handled.
+ * @param {string} file - A string referring to a file or folder.
+ * @param {string} [parentFolder='/'] - The previous folder that was handled before the current file.
+ * @returns {Array} An array of mapped files and folders into possible urls.
+ */
+function buildFolderSitemapEntries(sitemapEntries, file, parentFolder = '/') {
+    if (/^_/g.test(file)) {
+        return sitemapEntries;
+    }
+
+    if (file === 'index.js') {
+        return sitemapEntries.concat(parentFolder);
+    }
+
+    if (/.js$/g.test(file)) {
+        return sitemapEntries.concat(
```

Comment:
I prefer to create a new array and do it in a functional way. I just used `concat` because you suggested not creating a new array in cycles in previous code reviews 😄

Reply:
But concat creates new arrays. If you don't want to create new arrays, you may use `push`.